### PR TITLE
ci: Add Pipelines as Code CI

### PR DESCRIPTION
**Pull request by mbharatk**
### What does this PR do?
This will enable SPI Repo to easily configure Tekton Pipelines as Code CI for PR checks.
The goal of Pipelines as Code is to let you define your Tekton templates inside your source code repository and have the pipeline run and report the status of the execution when triggered by a Pull Request or a Push.

With this PR, the CI defined in [infra-deployments](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/spi/.tekton) will be moved to this repository

### CODE REVIEW

File: .tekton/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+spec:
+  params:
+    - name: repo_url
+      value: "{{repo_url}}"
+    - name: revision
+      value: "{{revision}}"
+  pipelineRef:
+          name: docker-build
+          bundle: quay.io/redhat-appstudio/build-templates-bundle:v0.1.4    
+  pipelineSpec:
```
Comment (Michkov) [inline .tekton/pull-request.yaml:18]: only one can be used -> pipelineRef or pipelineSpec

Comment (mbharatk) [reply .tekton/pull-request.yaml:15]: @Michkov made the necessary changes.

File: .tekton/pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        type: string
+    tasks:
+      - name: git-clone
+        params:
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+        taskRef:
+          kind: ClusterTask
```
Comment (Michkov) [inline .tekton/pull-request.yaml:29]: rather use remote tasks instead of ClusterTasks https://pipelinesascode.com/docs/guide/resolver/#remote-task-support

Comment (mbharatk) [reply .tekton/pull-request.yaml]: @Michkov changed, PTAL.

File: .tekton/push.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    pipelinesascode.tekton.dev/on-target-branch: "[main]"
+    pipelinesascode.tekton.dev/max-keep-runs: "2"
+spec:
+  params:
+    - name: git-url
+      value: 'https://github.com/redhat-appstudio/service-provider-integration-operator'
+    - name: revision
+      value: ${{revision}}
+    - name: output-image
+      value: 'quay.io/redhat-appstudio/service-provider-integration-operator:$(git-revision)'
```
[the remark this answers is not in the document]
Comment (mbharatk) [reply .tekton/push.yaml:17]: My bad, updating now.

**Comment by mbharatk** [comment]
@sparkoo could you please review this PR.

**Review by Michkov — APPROVED** [review]

File: .tekton/push.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    bundle: quay.io/redhat-appstudio/build-templates-bundle:b2cb5d5b21dc59d172379e639b336533bd8a8bf6
+  workspaces: 
+   - name: workspace 
+     volumeClaimTemplate: 
+       spec: 
+         accessModes: 
+           - ReadWriteOnce 
+         resources: 
+           requests: 
+             storage: 200Mi
```
Comment (skabashnyuk) [inline .tekton/push.yaml:33]: new line

Comment (mbharatk) [reply .tekton/push.yaml]: added.

**Comment by mbharatk** [comment]
@skabashnyuk Could you please take a look at this PR and give an LGTM.

**Comment by skabashnyuk** [comment]
@mbharatk can you please merge the latest main?

**Pull request merged**